src/main/java/seedu/address/ui/MatchListCard.java

```diff
@@ -76,19 +88,37 @@ private void initProperty(Property property) {
         propertySeller.setText(property.getSeller().getName().fullName);
         propertyPhone.setText(property.getSeller().getPhone().value);
         propertyEmail.setText(property.getSeller().getEmail().value);
-        property.getTags().stream()
-                .sorted(Comparator.comparing(tag -> tag.tagName))
-                .forEach(tag -> propertyTags.getChildren().add(new Label(tag.tagName)));
+
+        // Allow wrapping of labels
+        propertyPhone.maxWidthProperty().bind(maxWidth);
+        propertySeller.maxWidthProperty().bind(maxWidth);
+        propertyEmail.maxWidthProperty().bind(maxWidth);
+
+        List<Tag> tagList = new ArrayList<>(property.getTags());
+        tagList.sort(Comparator.comparing(tag -> tag.tagName));
+        for (Tag tag : tagList) {
+            propertyTags.getChildren().add(makeLabel(tag.tagName));
+        }
     }
 
     private void initBuyer(Buyer buyer) {
         buyerName.setText(buyer.getName().fullName);
         buyerPrice.setText('$' + buyer.getPrice().toString());
         buyerPhone.setText(buyer.getPhone().value);
         buyerEmail.setText(buyer.getEmail().value);
-        buyer.getTags().stream()
-                .sorted(Comparator.comparing(tag -> tag.tagName))
-                .forEach(tag -> buyerTags.getChildren().add(new Label(tag.tagName)));
+
+        List<Tag> tagList = new ArrayList<>(buyer.getTags());
+        tagList.sort(Comparator.comparing(tag -> tag.tagName));
+        for (Tag tag : tagList) {
+            buyerTags.getChildren().add(makeLabel(tag.tagName));
```

Comment:
Since you have added a `makeLabel` function, could we just modify the previous streaming solution?